### ISSUE TITLE
url '/', :index fails

### ISSUE THREAD

**Issue by wasnotrice** [shoes/shoes-deprecated#140]
Here's a minimal app using `url` that fails on current HEAD, OS X 10.7.1

``` ruby
class Gently < Shoes
  url '/', :index

  def index
    para "Do not go gently"
  end
end

Gently.app
```

the error looks like this:

```
wrong argument type UnboundMethod (expected Data)
gently.rb:9:in `app'
gently.rb:9:in `<main>'
/Users/eric/code/shoes/Shoes.app/Contents/MacOS/lib/shoes.rb:435:in `eval'
/Users/eric/code/shoes/Shoes.app/Contents/MacOS/lib/shoes.rb:435:in `visit'
/Users/eric/code/shoes/Shoes.app/Contents/MacOS/lib/shoes.rb:124:in `show_selector'
(eval):1:in `<main>'
```

**Comment by steveklabnik** [comment]
Hm. Very strange.

**Comment by derkosak** [comment]
I can confirm this on the latest gem published for windows. But the error is a bit different:

```
C:/prog/Ruby192/lib/ruby/gems/1.9.1/gems/green_shoes-1.0.282/lib/shoes/main.rb:19:in `app': undefined method `push'
 for nil:NilClass (NoMethodError)
        from teste.rbw:11:in `<main>'
```

**Comment by wasnotrice** [comment]
Even more succinct:

``` ruby
Shoes.app
```

fails with the same error, while

``` ruby
Shoes.app do; end
```

executes as expected.

**Comment by wasnotrice** [comment]
Another data point.

``` ruby
Shoes.app
```

runs fine using Policeman download version on OS X 10.6

**Comment by wasnotrice** [comment]
Fixed by cdbf79d7bd7231b49b35ccd883372154a27377af

**Comment by wasnotrice** [comment]
Premature closure. On linux, this kind of code produces a segfault, see http://librelist.com/browser//shoes/2011/9/15/segfault-with-shoes-build-on-the-nobody-knows-shoes-url-example/

**Comment by remear** [comment]
I'm also seeing the segfault. Ubuntu 11.04 32-bit

**Comment by Anonyfox** [comment]
Also see the segfault with Ubuntu 11.10, 64Bit. 
Adding an empty block to Shoes.app also opens a blank shoes window. 

Damn!

**Comment by ashbb** [comment]
Try out [this commit](https://github.com/ashbb/shoes/commit/b9e86ffd6f2e7438d5aae733e9b6d80ffcee37eb).

**Comment by steveklabnik** [comment]
Merged. Wooo.

**Comment by ghost** [comment]
I just downloaded the OS X version from your website, 5 minutes ago and I am getting this error. I am using the booklist example from the book and it works fine windows, haven't tried it on Linux, but I get an error every time. "wrong argument type UnboundMethod (expected Data)"
![screen shot 2013-08-02 at 1 13 06 pm](https://f.cloud.github.com/assets/1815397/902784/a4245558-fb9f-11e2-861d-b5046cdd2ec8.png)

Running OS X 10.8.4 with Ruby 2.0

**Comment by PragTob** [comment]
Hi there, thanks for the report. This issue currently is open in #236 - but is unlikely to get fixed here as work as halted on the old shoes. URLs work in the [new shoes4](https://github.com/shoes/shoes4) but that's not quite ready yet.

**Comment by ghost** [comment]
Sounds good. I'll check it out. Thanks for the info.

**Comment by ccoupe** [comment]
Bug fixed in Shoes 3.2.12. Tested with Windows, OSX and Linux.
